nomt/src/beatree/ops/update/tests.rs

```diff
@@ -0,0 +1,268 @@
+use lazy_static::lazy_static;
+use quickcheck::{Arbitrary, Gen, QuickCheck, TestResult};
+use rand::{Rng, SeedableRng};
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+use std::fs::File;
+
+use crate::io::{start_test_io_pool, IoPool};
+use crate::{
+    beatree::{
+        allocator::PageNumber,
+        branch::BRANCH_NODE_SIZE,
+        leaf::node::{body_size, LeafNode, MAX_LEAF_VALUE_SIZE},
+        ops::update::{preload_leaves, ChangedNodeEntry, LEAF_MERGE_THRESHOLD},
+        writeout, Tree,
+    },
+    io::PagePool,
+};
+
+const DB_INITIAL_CAPACITY: usize = 700;
+const CHANGESET_AVG_SIZE: usize = 300;
+const MAX_TESTS: u64 = 100;
+
+// Required to increase reproducibility
+lazy_static! {
+    static ref SEED: [u8; 16] = {
```

Comment:
You could also try reading an env first (like `NOMT_PRNG_TEST_SEED`) and in the custom `Drop` impl for your wrapper, when a test fails (`if thread::panicking()`) log the PRNG. This way you can reproduce the tests locally on your machine when they fail in CI (or on someone else's PC).

Reply:
That looks like a good suggestion. want to make a follow-up PR?

Reply:
That's a super nice idea, it would be very helpful ! Let us know if you want to work on that!

Reply:
Hello! It's been almost a month, and sadly, we really need to have reproducible bugs in CI. Thus, I had to implement something very similar to what you suggested! #570 

I didn't use the environment variable because I personally think it's easier to copy and paste the seed into the code. However, if you think it's valuable, there is still room for that update!

I tried to use `if thread::panicking()` inside a `drop`, but I discovered that `drop` is not performed on static elements (which makes sense). Thus, given how quickcheck lets you build and execute the tests, there were no other non-static struct which could easily implement the Drop trait.

Given those constraints, I went for the try and catch approach, which simply prints the seed if a panic occurs during the tests, and then it resumes the unwinding process.